fusion.py

```diff
@@ -0,0 +1,519 @@
+
+                # Donor sequence may not align to some isoforms
+                if bounds is None:
+                    logging.debug('Donor sequence did not align to reference transcripts')
+                    continue
+
+            except KeyError:
+                continue
+
+            # Find the position of the transcript where the predicted donor sequence starts
+            in_frame_adjustment = scan_frame(bounds.rstart)
+            if in_frame_adjustment is None:
+                continue
+
+            # Slice the donor transcript such that the first base is in-frame
+            # May have an overhang base
+            donor_start = bounds.qstart + in_frame_adjustment
+            in_frame_donor_seq = junctionSeq1[donor_start:]
+            in_frame_donor_seq_len = len(in_frame_donor_seq)
+
+            print('JUNCTIONSEQ1')
+            print(in_frame_donor_seq)
+
+            if acceptor_name in gene_transcripts:
+                acceptor_transcript_ids = gene_transcripts[acceptor_name]
+
+            # If we don't know the acceptor transcripts, then skip fusion
+            else:
+                logging.debug('Could not find acceptor transcript ID')
+                break
+
+            # Now iterate over all possible acceptor transcript sequences
+            start2 = int(call.start2)
+            for acceptor_transcript_id in acceptor_transcript_ids:
+                junctionSeq2 = call.junctionSeq2
+
+                # Predict acceptor sequence if none was provided
+                if exons and exons[acceptor_transcript_id] and junctionSeq2 == '.':
+                    junctionSeq2 = get_acceptor_junction(start2, exons[acceptor_transcript_id])
+
+                if (junctionSeq1 == junctionSeq2 == '.') or (not junctionSeq1 and not junctionSeq2):
+                    logging.debug("Could not generate fusion sequence")
+                    continue
+
+                acceptor_in_frame_adjustment = None
+                try:
+                    # Get reference coding transcript for acceptor gene
+                    acceptor_transcript = transcriptome[acceptor_transcript_id]
+
+                    # Find the position of the transcript where the predicted acceptor sequence starts
+                    acceptor_bounds = align_filter(acceptor_transcript, junctionSeq2, 'acceptor')
+
+                    # If the acceptor sequence could not be found, move onto the next transcript
+                    if acceptor_bounds is None:
+                        continue
+
+                    acceptor_in_frame_adjustment = scan_frame(acceptor_bounds.rstart)
+                    if acceptor_in_frame_adjustment is None:
+                        continue
+
+                # We can still generate a fusion transcript without a reference for the acceptor gene
+                except KeyError:
+                    pass
+
+                # Figure out if frame-shift occurred as a result of the fusion
+                acceptor_frame_shift = False
```

Comment:
Is that logic in right now or is this a future addition? It didn't look like that variable was used anywhere but i could be wrong.